### PR TITLE
fix check for missing installations

### LINKED ISSUE #236: `check_missing_installations.sh` fails to capture missing installations correctly
Seems that `check-missing-installations.sh` in combination with `EESSI-pilot-install-software.sh` fails to capture missing installations correctly:
```bash
[foo ~]$ cat fake-stack.yml
easyconfigs:
  - EasyBuild-4.6.2.eb
  - xxd-8.2.4220-GCCcore-10.2.0.eb
  - Nextflow-22.10.1.eb
[foo ~]$ ml --show-hidden av EasyBuild/4.6.2

--------------------------------------------------------------------------------------------------------- /cluster/modulefiles/all ----------------------------------------------------------------------------------------------------------
   EasyBuild/4.6.2 (H)

  Where:
   H:  Hidden Module

Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching any of the "keys".


[foo ~]$ ml --show-hidden av xxd/8.2.4220-GCCcore-10.2.0

--------------------------------------------------------------------------------------------------------- /cluster/modulefiles/all ----------------------------------------------------------------------------------------------------------
   xxd/8.2.4220-GCCcore-10.2.0

Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching any of the "keys".


[foo ~]$ ml --show-hidden av Nextflow/22.10.1
No modules found!
Use "module spider" to find all possible modules.
Use "module keyword key1 key2 ..." to search for all possible modules matching any of the "keys".


[foo ~]$ eb --easystack fake-stack.yml --missing --experimental
== Temporary log file in case of crash /tmp/eb-7g6d1dfs/easybuild-_5z6vudw.log
== found valid index for /cluster/software/EasyBuild/4.7.0/easybuild/easyconfigs, so using it...

No missing modules!

== found valid index for /cluster/software/EasyBuild/4.7.0/easybuild/easyconfigs, so using it...
== found valid index for /cluster/software/EasyBuild/4.7.0/easybuild/easyconfigs, so using it...

No missing modules!

== found valid index for /cluster/software/EasyBuild/4.7.0/easybuild/easyconfigs, so using it...

2 out of 3 required modules missing:

* Java/11.0.16 (Java-11.0.16.eb)
* Nextflow/22.10.1 (Nextflow-22.10.1.eb)

== Temporary log file(s) /tmp/eb-7g6d1dfs/easybuild-_5z6vudw.log* have been removed.
== Temporary directory /tmp/eb-7g6d1dfs has been removed.
```
For each line (easyconfig file) in the easystack file the test is run (probably fine), which then prints several `No missing modules!` lines (also fine), but `check_missing_installations.sh` just greps for this line in the output of the command `eb --easystack fake-stack.yml --missing --experimental` and therefore does not catch the missing installation.

### PR DESCRIPTION
**Pull request by trz42**
fixes #236 

Added comments to explain the needed change.

Should we add a test for this?

### CODE REVIEW

**Comment by trz42** [comment]
Converted to a **draft** because it is only necessary if we use EB v4.7.0 or newer.

**Comment by trz42** [comment]
Changed script works for EB v4.5.1 (current latest EB version in EESSI pilot).

Added a test that captures a missing package.

The tests may fail if EB v4.7.0 or newer is used AND the format of the easystack file is not updated.

File: check_missing_installations.sh [hunk 1]
```diff
@@ -27,5 +27,21 @@ eb_missing_out=$LOCAL_TMPDIR/eb_missing.out
...
+# `No missing modules!`
+# or
+# `2 out of 3 required modules missing:`
+# depending on the result of the assessment. Hence, we need to check if the
+# output does not contain any line with ` required modules missing:`
+
+grep " required modules missing:" ${eb_missing_out} > /dev/null
+
+# we need to process the result (from finding `No missing modules` to NOT finding
+# ` required modules missing:` and no other error happened)
```
Comment (boegel) [inline check_missing_installations.sh:42]: @trz42 This comment looks like a TODO description, can be removed?

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**